libsolidity/analysis/TypeChecker.cpp

```diff
@@ -4024,18 +4041,43 @@ void TypeChecker::endVisit(Literal const& _literal)
 					// This was already validated in visit(Literal) but the error is not fatal.
 					if (!mantissa || !exponent)
 						solAssert(!m_errorReporter.errors().empty());
-					else if (
-						!mantissa->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)) ||
-						!exponent->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(1))
-					)
-						// TODO: Is this triggered when the argument is out of range? Test.
-						parameterTypeMessage = "The type of the literal cannot be converted to the parameters of the suffix function.";
+					// TODO: Is this triggered when the argument is out of range? Test.
+					else if (!mantissa->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)))
+						parameterTypeMessage = fmt::format(
+							"Type {} of the mantissa cannot be converted to type {} of the suffix function parameter.",
+							mantissa->mobileType()->toString(true),
+							suffixFunctionType->parameterTypes().at(0)->toString(true)
+						);
+					else if (!exponent->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(1)))
+					{
+						if (!parameterTypeMessage.has_value())
+							parameterTypeMessage = "";
+						parameterTypeMessage.value() += fmt::format(
+							"Type {} of the exponent cannot be converted to type {} of the suffix function parameter.",
+							exponent->mobileType()->toString(true),
+							suffixFunctionType->parameterTypes().at(1)->toString(true)
+						);
+					}
 				}
 			}
 			else if (suffixFunctionType->parameterTypes().size() == 1)
 			{
 				if (!literalType->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)))
-					parameterTypeMessage = "The type of the literal cannot be converted to the parameter of the suffix function.";
+				{
+					if(literalRationalType)
+					{
+						if (literalRationalType->mobileType())
+							parameterTypeMessage = fmt::format(
+								"The type {} of the literal cannot be converted to the type {} of the suffix function parameter.",
+								literalRationalType->mobileType()->toString(true),
+								suffixFunctionType->parameterTypes().at(0)->toString(true)
+							);
+						else
+							parameterTypeMessage = "The value of the literal is too large and cannot be converted to any available type.";
+					}
+					else
+						parameterTypeMessage = "The type of the literal cannot be converted to the parameter of the suffix function.";
```

Comment:
Let's reword this one too to match:
```suggestion
						parameterTypeMessage = "The literal is out of range of parameters of any possible suffix function.";
```

Reply:
What about this one? It's resolved but you did not change it or leave a comment saying why.

Reply:
Actually, after looking at tests I see why it's better left as is. We went to mention the range only for integers.

I'd still reword it slightly to match the other messages though:
```suggestion
						parameterTypeMessage = "The literal cannot be converted to the type accepted by the suffix function.";
```

Reply:
Sorry, I probably missed it while applying the suggestions and mistakenly marked as resolved without discussing it.
I will apply the suggestion.